### DISCUSSION TITLE
Tensor shape when predicting from Clustered Multitask GP Regression

### DISCUSSION THREAD

**Discussion by tcristea** [cornellius-gp/gpytorch#1923] [Q&A] [answered]
Hi,

I've run this example:
[https://github.com/cornellius-gp/gpytorch/blob/master/examples/07_Pyro_Integration/Clustered_Multitask_GP_Regression.ipynb](url)
When in the 6th code cell, the fragment:
 ```# Set into eval mode and make predictions  
model.eval()  
with torch.no_grad():  
    test_x = torch.linspace(0, 1, 51)
    predictions = model.likelihood(model(test_x)).rsample()
print(predictions.shape)  # num_samples x num_data x num_tasks
 ```
the shape of prediction tensor is [10, 51, 4], i.e., num_data = 51 (size of test_x), num_tasks = 4 (set to the number of train_y features), while num_samples is always comes out as 10.
The value of num_samples (= 10) is set by this statement **model.likelihood(model(test_x))**.
Where is the "10" coming from? I've read the code but I'm not any wiser.

Best regards,
TC

**Comment by wjmaddox** [comment, marked as answer]
It's this setting: https://docs.gpytorch.ai/en/stable/settings.html#gpytorch.settings.num_likelihood_samples

Basically when latent GPs are evaluated, gpytorch automatically draws several samples for usage in MC estimation over the latent GP.

**Reply by tcristea** [reply]
Thank you!
